pageserver/src/tenant/storage_layer.rs

```diff
@@ -277,77 +314,30 @@ impl Default for LayerFringe {
     }
 }
 
-impl Ord for ReadableLayerDescOrdered {
-    fn cmp(&self, other: &Self) -> Ordering {
-        let ord = self.0.get_lsn_ceil().cmp(&other.0.get_lsn_ceil());
-        if ord == std::cmp::Ordering::Equal {
-            self.0
-                .get_lsn_floor()
-                .cmp(&other.0.get_lsn_floor())
-                .reverse()
-        } else {
-            ord
-        }
-    }
-}
```

Comment:
This was a bit difficult to see how this was transformed with the other changes to what we now have in `impl Ord for ReadDesc`, but looks the same as this was. Would we lose something if the ordering was not shuffled?

Reply:
Not sure what you mean by shuffling here, but yeah the new code is equivalent with the previous one in terms of read ordering.

This ordering defines which will read will get picked next. We pick the read that starts at the highest lsn. If there are multiple reads that which end at the same max lsn, we pick the longest read in terms of lsn. This is sort of arbitrary come to think of it. I think a queue would work just as well here.

Reply:
Are you suggesting I move it back to it's original place? Happy to do it, but unsure if that's what you're implying.

Reply:
Moved the `Ord` impl back to the old place to reduce the diff: https://github.com/neondatabase/neon/pull/7215/commits/ab8ce836f377aa42e01507e55c6b5509d16ed6bc